File: rai_rai_raku_raku.py
#!/usr/bin/python3
import discord
import random
import time
import sys

tokenFile = open("token.txt")
token = tokenFile.read()

spam = False

charset = "jN2curS8H8ywvvBWnewPWyYnKTcDhYKJf1WMyzitUuBpZDVtqQ9K7HjsUdlXeD8kda08QPJ0HPrDzx7iwwtAaAGSg5eeN22pctNZA5qrHEd8FsomXM38qvHQXYLVTUqGyKEcwUjtFx7VNeIyHQ7WUNZmwCnfDiQjGnSo7ZDDA7lO1UHD25H1zki8BXyj89RMvGkRYOQV3EPO97pumRUfaq6UGdnpujZoD0tVarNPOGFJLbgDKv0znjRDr5cF9nw"

channel_id = 0

my_client = discord.Client()

async def process_commands():
    print("[INFO]: The bot is now acting as a mouthpiece")
    await my_client.wait_until_ready()
    while not my_client.is_closed():
        command = input(">")
        if command == "stop":
            print("Stopping the bot.")
            await my_client.close()
            break
        if command.startswith("send"):
            global channel_id
            channel = my_client.get_channel(channel_id)
            await channel.send(command.split("#|")[1])
        if command.startswith("set_channel"):
            channel_id = int(command.split(" ")[1])
        #end
    #end
#end

@my_client.event
async def on_ready():
    print(f"The bot have logged in as {my_client}")
    
    if len(sys.argv) > 1:
        if sys.argv[1] == "remote":
            my_client.loop.create_task(process_commands())
        else:
            print("[INFO]: The bot is now operating without remote control.")
    else:
        print("[INFO]: The bot is now operating without remote control.")
#end

@my_client.event
async def on_message(message: discord.Message):
    if message.author == my_client.user:
        return
    #end
    
    if message.author.id == 650439182204010496 or message.author.id == 690265771955585029:
        command = message.content
        if command.startswith("$$$send"):
            global channel_id
            channel = my_client.get_channel(channel_id)
            await channel.send(command.split("#|")[1])
        if command.startswith("$$$set_channel"):
            channel_id = int(command.split(" ")[1])
        if command.startswith("$$$reply"):
            channel = my_client.get_channel(channel_id)
            replyMessage = await channel.fetch_message(int(command.split(" ")[1]))
            sendMessage = command.split("#|")[1]
            await replyMessage.reply(sendMessage)
        #end
    #end

    if message.content.startswith("$$$stop_spamming$$$"):
        global spam
        spam = False
    #end

    if message.content.startswith("$$$start_spamming$$$"):
        try:
            prefix = message.content.split(" ")[1]
        except:
            prefix = ""
        #end
        
        try:
            send_message = message.content.split("||")[1]
            prefix = ""
        except:
            send_message = ""
            for i in range(0, 50):
                char = random.choice(charset)
                send_message = send_message + char
            #end
        #end
        
        try:
            delay = float(message.content.split("||")[2])
        except:
            delay = 1
        #end
        
        spam = True
        while spam:
            await message.channel.send(prefix + " " + send_message)
            time.sleep(delay)
        #end
    #end
#end

my_client.run(token)

File: rai_rai_raku_raku.js
const
{
    Client,
    Intents,
    DMChannel,
    TextChannel,
    Collection,
    MessageManager,
    CategoryChannel
} = require("discord.js")
const { token } = require("./token.json")
const Str = require("@supercharge/strings")

const bot = new Client
(
    { 
        intents:
        [
            Intents.FLAGS.GUILDS,
            Intents.FLAGS.GUILD_MESSAGES,
            Intents.FLAGS.DIRECT_MESSAGES,
        ]
    }
)

let channelToSend = "0"
let spam = false


function onReady()
{
    console.log("[INFO]: The client has logged on as " + bot.user.username + "#" + bot.user.discriminator)
}



function spam_f(channel, prefix, msg, masterMessage)
{
    if (spam === false)
    {
        return
    }
    
    channel.send(prefix + " " + msg).then(() =>
    {
        setTimeout(() => 
        {
            spam_f(channel, prefix, msg, masterMessage);
        }, 1500);
    }).catch((error) => 
    {
        masterMessage.reply("I cannot continue spamming the target because of this error.\n" + 
                            "```" +
                            error +
                            "```")
    })
}



async function onMessageCreate(message)
{
    if (DMChannel.prototype.isPrototypeOf(message.channel))
    {
        console.log("(DM of " + message.channel.recipient.username + "): " + message.author.username + ": " + message.content)
    } else {
        console.log("(#" + message.channel.name + "): " + message.author.username + ": " + message.content)
    }
    
    if (message.author.id == bot.user.id)
    {
        return;
    }
    
    // Check for trademark violations
    if (message.author.id != "690265771955585029" && message.author.id != "725811783012450306")
    {
        part1 = message.content.toLowerCase();
        
        if (part1.includes("disappointment"))
        {
            let part2 = part1.split("disappointment")[1];
            if (part2.includes("immeasurable"))
            {
                let part3 = message.content.split("immeasurable")[1];
                if (part3.includes("day"))
                {
                    let part4 = message.content.split("day")[1];
                    if (part4.includes("ruined"))
                    {
                        message.reply("\"My disappointment is immeasurable and my day is ruined\" and all variations are registered trademarks of 0051 as of October 2021. Please cease and desist from all current and future uses. Thank you in advance for your cooperation.");
                        //message.channel.send("<@690265771955585029> Your trademark has been violated.")
                    }
                }
            }
        }
    }
    
    // Special control commands
    if (message.author.id == "650439182204010496" || message.author.id == "690265771955585029")
    {
        var command = message.content
        if (command.startsWith("$$$send"))
        {
            bot.channels.fetch(channelToSend).then(channel => {
                channel.send(command.split("#|")[1])
            })
        } else if (command.startsWith("$$$set_channel"))
        {
            channelToSend = command.split(" ")[1]
            console.log("Channel is now set to " + channelToSend)
        } else if (command.startsWith("$$$reply"))
        {
            bot.channels.fetch(channelToSend).then(channel => {
                channel.messages.fetch(command.split(" ")[1]).then(msgToReplyTo => {
                    msgToReplyTo.reply(command.split("#|")[1])
                })
            })
        }
    }
    
    if (message.content.startsWith("$$$stop_spamming$$$"))
    {
        spam = false
        return
    }
    
    if (message.content.startsWith("$$$start_spamming$$$"))
    {
        spam = true
        
        let prefix = message.content.split(" ")[1]
        if (prefix === undefined || prefix === "||")
        {
            prefix = ""
        }
        
        let msg = message.content.split("||")[1]
        if (msg === undefined)
        {
            msg = "i like cute girls"
        }
        
        setTimeout(() => {
            spam_f(message.channel, prefix, msg)
        }, 1500)
    }
    
    if (message.content.startsWith("$$$start_spamming_dm$$$"))
    {
        spam = true
        
        let channelId = message.content.split(" ")[1]
        
        let msg = message.content.split("||")[1]
        if (msg == undefined)
        {
            msg = "i like cute girls"
        }
        
        
        
        message.guild.members.fetch(channelId).then((channel) => {
            console.log("Spamming user " + channel.user.username + "'s DM.")
            
            spam_f(channel, "<@" + channelId + ">", msg, message)
            
        }).catch(error => {
            message.reply("The specified user could not be found.");
            console.log(error)
        })
    }
    
    if (message.content.startsWith("$$$nuke_channel$$$"))
    {
        var position = message.channel.position;
        message.channel.clone().then((clonedChannel) => {
            clonedChannel.setPosition(position)
        })
        message.channel.delete()
    }
    
    if (message.content.startsWith("$$$nuke_server$$$"))
    {
        message.channel.guild.channels.fetch().then((channels) => {
            channels.each((channel) => {
                if (CategoryChannel.prototype.isPrototypeOf(channel))
                {
                    return;
                }
                
                var position = channel.position;
                
                channel.clone().then((clonedChannel) => {
                    clonedChannel.setPosition(position)
                    channel.delete().catch(() => {
                        clonedChannel.delete().catch(console.error)
                    })
                }).catch(console.error)
                
            })
        }).catch(console.error)
    }
}

bot.on("ready", onReady)
bot.on("messageCreate", onMessageCreate)

bot.login(token)
